modules/ROOT/pages/creating-a-custom-policy.adoc

```diff
@@ -0,0 +1,292 @@
+
+The workflow to create custom policies for Mule 3.9 and earlier in Anypoint API Manager has not changed. Continue to use the process of developing the policy, packaging the custom policy, uploading it to Exchange, and applying the policy to an API through Anypoint API Manager as described in the Anypoint API Manager documentation. 
+
+The workflow for creating custom policies for Anypoint Flex Gateway and Mule 4 runtime is done in Exchange. The policy is then applied to an API in Anypoint Manager.
+
+The following steps are the high-level workflow for publishing custom policies for Anypoint Flex Gateway and Mule 4:
+
+. Create policy definition files
+. Create implementation files
+. Publish custom policy asset to Exchange
+. Add implementation files
+. Apply the policy to an API from Anypoint API Manager
+
+== Before You Begin
+
+Before you can publish a custom policy for Flex Gateway or Mule 4, ensure you have administrator or contributor permission for the Anypoint Platform.
+
+
+[[create-policy-definition_files]]
+== Create Policy Definition Files
+
+Policy definition files must be created before you can upload the policy implementation. The required files are JSON and YAML. The JSON file contains the implementation of the policy using JSON schema language. Use the JSON example to construct the custom policy configuration. The YAML file defines the configurable parameters of the policy. Anypoint API Manager uses this file to render the UI to display user inputs for the policy.
+
+Copy the custom policy definition examples or create the files on you local system.
+
+=== Create the JSON Schema File
+
+The JSON schema file specifies the interface for the policy by identifying which properties are required or supported to configure the policy. The file also includes more annotations to refine the type for each property that enables some fields to perform a special behavior in the Anypoint platform. For example, Dataweave validations or secure handling of secrets.
+
+The JSON file must include the following fields:
+
+* Top-level @context
++
+References the vocabulary used for the extra type annotations. The aliases used for namespacing the vocabulary are a optional, but common names could be `security` or `config`.
+* Top-level title
+* $schema with a link to the JSON Schema spec (see example)
+* Root type of `object`
+* Top-level description
+* Each property must also include:
++
+** Title
++
+The title is used by API Manager UI to render the field name.
+** Description
++
+The description is used by API Manager UI to describe the field.
+** Type 
++
+Only primitive types are supported, for example `number`, `string`, `object`, `boolean`, `array`, `null`.
+
+Each property can include the following:
+
+* Validations 
```

Comment:
I think validations remained old. We could instead mention:
- `format`: options are `dataweaveExpression` and `ipRange`
- `default`: allows the user to enter a default value which will be rendered in the UI

Reply:
fixed